kubernetes/watcher.go

```diff
@@ -182,6 +202,46 @@ func NewNamedWatcher(name string, client kubernetes.Interface, resource Resource
 	return w, nil
 }
 
+// NewMetadataWatcher initializes a metadata-only watcher client to provide an events handler for
+// resource from the cluster (filtered to the given node).
+// Event handlers defined on this watcher receive PartialObjectMetadata resources.
+func NewMetadataWatcher(
+	client kubernetes.Interface,
+	metadataClient metadata.Interface,
+	gvr schema.GroupVersionResource,
+	opts WatchOptions,
+	indexers cache.Indexers,
+	transformFunc cache.TransformFunc,
+) (Watcher, error) {
+	return NewNamedMetadataWatcher("", client, metadataClient, gvr, opts, indexers, transformFunc)
```

Comment:
nit: let's note that passing an empty string for a name here, will result in a workqueue without any metrics exposed which we might want to utilise in the future

Reply:
Added a note to the docstring.